app/modules/encounters/resources.py

```diff
@@ -107,20 +107,50 @@ def post(self, args):
 
         # if we get here, edm has made the encounter, now we create & persist the feather model in houston
 
+        # will contain EncounterAssets objects to join to assets (dont load assets themselves)
+        asset_refs = []
+        if 'assets' in data and isinstance(data['assets'], list):
+            from app.modules.encounters.models import EncounterAssets
+
+            for asset_data in data['assets']:
+                if isinstance(asset_data, dict) and 'guid' in asset_data:
+                    # note: if an invalid asset guid is provided, foreign key contstraint error will be thrown when persisting
+                    asset_ref = EncounterAssets(
+                        encounter_guid=result_data['id'], asset_guid=asset_data['guid']
+                    )
+                    asset_refs.append(asset_ref)
+
         context = api.commit_or_abort(
-            db.session, default_error_message='Failed to create a new Encounter'
+            db.session, default_error_message='Failed to create a new houston Encounter'
         )
-        with context:
-            # TODO other houston-based relationships: orgs, projects, etc
-            owner_guid = None
-            if current_user is not None:
-                owner_guid = current_user.guid
-            encounter = Encounter(
-                guid=result_data['id'],
-                version=result_data.get('version', 2),
-                owner_guid=owner_guid,
+        try:
+            with context:
+                # TODO other houston-based relationships: orgs, projects, etc
+                owner_guid = None
+                pub = True  # legit? public if no owner?
+                if current_user is not None:
+                    owner_guid = current_user.guid
+                    pub = False
+                encounter = Encounter(
+                    guid=result_data['id'],
+                    version=result_data.get('version', 2),
+                    owner_guid=owner_guid,
+                    public=pub,
+                )
+                encounter.assets = asset_refs
+                db.session.add(encounter)
+        except Exception as ex:
```

Comment:
_this_ is the point i wish to discuss on friday.  there are a few permutations of failure cases i try to recover from here.  the one i need some closure on is when the **edm _successfully_ creates** its object (the "primary" one), and **houston fails** to create/persist its feather equivalent right after that.
currently i am leaning toward cleaning up by calling back to edm to delete what it just made (see line 151), but there are other ways to look at this -- and probably other permutations i havent considered.

Reply:
discussion with @colinwkingen concluded to go with edm-object delete as recovery behavior here.  basically, cleanup/undo what has happened and then fail on the api.  going to add that asap.